pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go

```diff
@@ -297,3 +303,106 @@ func (impl AppStoreDeploymentFullModeServiceImpl) createInArgo(chartGitAttribute
+	if err != nil {
+		impl.logger.Errorw("error in unmarshal", "err", err)
+		return installAppVersionRequest, err
+	}
+	valuesMap := make(map[string]map[string]interface{})
+	valuesMap[installAppVersionRequest.AppStoreName] = dat
+	valuesByte, err := json.Marshal(valuesMap)
+	if err != nil {
+		impl.logger.Errorw("error in marshaling", "err", err)
+		return installAppVersionRequest, err
+	}
+	valuesConfig := &util.ChartConfig{
+		FileName:       appStoreBean.VALUES_YAML_FILE,
+		FileContent:    string(valuesByte),
+		ChartName:      installAppVersionRequest.AppStoreName,
+		ChartLocation:  acdAppName,
+		ChartRepoName:  installAppVersionRequest.GitOpsRepoName,
+		ReleaseMessage: fmt.Sprintf("release-%d-env-%d ", installAppVersionRequest.AppStoreVersion, installAppVersionRequest.EnvironmentId),
+	}
+	gitOpsConfigBitbucket, err := impl.gitOpsRepository.GetGitOpsConfigByProvider(util.BITBUCKET_PROVIDER)
+	if err != nil {
+		if err == pg.ErrNoRows {
+			gitOpsConfigBitbucket.BitBucketWorkspaceId = ""
+		} else {
+			impl.logger.Errorw("error in fetching gitOps bitbucket config", "err", err)
+			return installAppVersionRequest, err
+		}
+	}
+	commitHash, err := impl.gitFactory.Client.CommitValues(valuesConfig, gitOpsConfigBitbucket.BitBucketWorkspaceId)
+	if err != nil {
+		impl.logger.Errorw("error in git commit", "err", err)
+		return installAppVersionRequest, err
+	}
+	installAppVersionRequest.GitHash = commitHash
+	return installAppVersionRequest, nil
+}
+
+func (impl AppStoreDeploymentFullModeServiceImpl) UpdateRequirementYaml(installAppVersionRequest *appStoreBean.InstallAppVersionDTO, appStoreAppVersion *appStoreDiscoverRepository.AppStoreApplicationVersion) error {
+	acdAppName := fmt.Sprintf("%s-%s", installAppVersionRequest.AppName, installAppVersionRequest.EnvironmentName)
+	dependency := appStoreBean.Dependency{
+		Name:       appStoreAppVersion.AppStore.Name,
+		Version:    appStoreAppVersion.Version,
+		Repository: appStoreAppVersion.AppStore.ChartRepo.Url,
+	}
+	var dependencies []appStoreBean.Dependency
+	dependencies = append(dependencies, dependency)
+	requirementDependencies := &appStoreBean.Dependencies{
+		Dependencies: dependencies,
+	}
+	requirementDependenciesByte, err := json.Marshal(requirementDependencies)
+	if err != nil {
+		return err
+	}
+	requirementDependenciesByte, err = yaml.JSONToYAML(requirementDependenciesByte)
+	if err != nil {
+		return err
+	}
+	requirmentYamlConfig := &util.ChartConfig{
+		FileName:       appStoreBean.REQUIREMENTS_YAML_FILE,
+		FileContent:    string(requirementDependenciesByte),
+		ChartName:      appStoreAppVersion.AppStore.Name,
+		ChartLocation:  acdAppName,
+		ChartRepoName:  installAppVersionRequest.GitOpsRepoName,
+		ReleaseMessage: fmt.Sprintf("release-%d-env-%d ", appStoreAppVersion.Id, installAppVersionRequest.EnvironmentId),
+	}
+	gitOpsConfigBitbucket, err := impl.gitOpsRepository.GetGitOpsConfigByProvider(util.BITBUCKET_PROVIDER)
+	if err != nil {
+		if err == pg.ErrNoRows {
```

Comment:
this code should be inside GetGitOpsConfigByProvider

Reply:
GetGitOpsConfigByProvider has only its global config services